src/lib/OpenCvProvider.js

```diff
@@ -32,12 +35,12 @@ export const OpenCvProvider = ({ openCvPath, children, onLoad }) => {
     // https://medium.com/code-divoire/integrating-opencv-js-with-an-angular-application-20ae11c7e217
     // https://stackoverflow.com/questions/56671436/cv-mat-is-not-a-constructor-opencv
     moduleConfig.onRuntimeInitialized = handleOnLoad
-    window.Module = moduleConfig
+    window.Module = { ...moduleConfig, ...openCvConfig }
 
-    const generateOpenCvScriptTag = () => {
+    const generateOpenCvScriptTag = async () => {
       const js = document.createElement('script')
       js.id = scriptId
-      js.src = openCvPath || 'https://docs.opencv.org/3.4.13/opencv.js'
+      js.src = (typeof openCvPath === 'function') ? await openCvPath() : openCvPath || 'https://docs.opencv.org/3.4.13/opencv.js'
```

Comment:
why would you pass a promise as the openCvPath function? Doesn't make sense to asynchronously wait for a string. What's/was your use case?